Http Range requests support in downloads

lenz1111 opened the pull request:
Http range requests support is required for video preview

luckydonald committed:
Upload progress bar now displays informations about uploades size (12kB of 3MB) and the current speed.

luckydonald committed:
Fixed translations, indentation and renamed the .percents class to .speed class.

luckydonald committed:
Updated and optimized Mobile Layout as well.

luckydonald committed:
...in case the text gets to big, don't wrap to new line. Instead just cut it.

luckydonald committed:
Added remaining time displaying, moved Information to tooltip, removed now unneeded speed div

luckydonald committed:
Fixed Positioning, fixed formatting for both mobile and desktop.
Tooltips still frustrate me-
And the updated code with tabs nicely. yay.

luckydonald committed:
Updated String formatting. Label font-weight was already normal.

luckydonald committed:
Fixed "Any minute now" staying because of undefined buffer. Also increased buffer size to 20 to be less jittery.

luckydonald committed:
fixed browser showing the tipsy tooltip as default tooltip twice.

luckydonald committed:
Changes for scrutinizer. Yay!

luckydonald committed:
Optimized time formatting, optimized time smoothing.

pellaeon committed:
Handle case when breadcrumb and filelist item overlap
and both receive a drop.

Build breadcrumb over and out callback in filelist.js and
pass them as options to utilize local selectors of filelist

Re-enable td elements after drop to bread crumb

Fix# drop handler still being called after enable,
it seems that waiting for a short time before re-enabling solves the
problem

* Add explanations for setTimeout re-enable td.filename drop handler

Commit:
Allow multiple custom JS files

Commit:
Add links to load GDrive/Dropbox configuration pages

Commit:
Change colour of input fields instead of printing 'Access granted'

RobinMcCorkell committed:
Move Dropbox/GDrive link to icon

icewind1991 committed:
factor out redis configuration into it's own factory

mmattel committed:
Adding mimetypes for m4a and m4b

LukasReschke committed:
Read only once

This can lead to race conditions otherwise. Also depracted hasKey

ChristophWurst committed:
token based auth
* Add InvalidTokenException
* add DefaultTokenMapper and use it to check if a auth token exists
* create new token for the browser session if none exists
hash stored token; save user agent
* encrypt login password when creating the token

ChristophWurst committed:
invalidate (delete) session token on logout
add 'last_activity' column to session tokens and delete old ones via a background job

ChristophWurst committed:
Check if session token is valid and log user out if the check fails
* Update last_activity timestamp of the session token
* Check user backend credentials once in 5 minutes

ChristophWurst committed:
Add controller to generate client tokens

ChristophWurst committed:
Add index on 'last_activity'

add token type column and delete only temporary tokens in the background job

debounce token updates; fix wrong class import

ChristophWurst committed:
Add token auth for OCS APIs

ChristophWurst committed:
Add fallback to allow user:token basic auth

ChristophWurst committed:
increase version number to trigger db migration

ChristophWurst committed:
fix setup

ChristophWurst committed:
fix LoginController unit tests

ChristophWurst committed:
Fix existing tests

ChristophWurst committed:
increase token column width

add some range to time() assertions

ChristophWurst committed:
try apache auth too

ChristophWurst committed:
pass in $request on OCS api

lenz1111 committed:
Http Range requests support in downloads

lenz1111 committed:
Http Range requests support in downloads

lenz1111 committed:
Http Range requests support in downloads

lenz1111 committed:
Http Range requests support in downloads

lenz1111 committed:
Http Range requests support in downloads

lenz1111 committed:
Http Range requests support in downloads

lenz1111 committed:
cleanup

lenz1111 committed:
cleanup

lenz1111 committed:
cleanup

lenz1111 committed:
cleanup

DeepDiver1975 commented:
hmmm ... the video player should use the webdav url - this interface is already supporting streaming

timmmmmey commented:
> PVince81 commented 10 days ago
At first I'd have said the video player needs to be adjusted to use the public Webdav URL instead, which supports Range requests. However, this endpoint has some issues in some browsers due to the fact that it requires either a username to be passed in the URL or an "Authorization" header to be set, and both aren't supported by the video API.

See owncloud/core#21787 (comment)

>

PVince81 commented:
@DeepDiver1975 yeah, sad thing

DeepDiver1975 commented:
let's see if we can clean up that mess in 9.1 ....

PVince81 commented:
@DeepDiver1975 the only way I see is to have a new better endpoint that replaces publicwebdav

PVince81 commented:
@DeepDiver1975 can we whip the bot ?

lenz1111 commented:
I already signed (and sent) contributors agreement...

DeepDiver1975 commented:
> I already signed (and sent) contributors agreement...

When did you send the agreement and to whom? THX

lenz1111 commented:
@DeepDiver1975 Today, to Carlos: cml@owncloud.com

DeepDiver1975 commented:
> @DeepDiver1975 Today, to Carlos

THX *I removed the email from your comment*

ChristophWurst committed:
add unit tests for all new classes

ChristophWurst committed:
add IClientService to DI container

ChristophWurst committed:
fix mock builder for old phpunit versions

ChristophWurst committed:
PHPDoc and other minor fixes

LukasReschke committed:
Use proper path

Fixes https://github.com/owncloud/core/issues/24407

LukasReschke commented on lib/private/Files/View.php:
```diff
@@ -425,6 +425,32 @@ public function readfile($path) {
+		@ob_end_clean();
+		$handle = $this->fopen($path, 'rb');
+		if ($handle) {
+			if (fseek($handle, $from) == 0) {
```
`===`

LukasReschke commented on lib/private/Files/View.php:
```diff
@@ -425,6 +425,32 @@ public function readfile($path) {
+			if (fseek($handle, $from) == 0) {
+			    $chunkSize = 8192; // 8 kB chunks
+			    while (!feof($handle) && ftell($handle) <= $to) {
+				$len = $to-ftell($handle)+1 > $chunkSize ? $chunkSize : $to-ftell($handle)+1;
```
Move ` $to-ftell($handle)+1` into variable

ChristophWurst committed:
add auth integration tests

ChristophWurst committed:
catch possible SessionNotAvailableExceptions

lenz1111 committed:
code optimization

lenz1111 committed:
code optimization

lenz1111 committed:
code optimization

lenz1111 committed:
code optimization

rullzer committed:
Move OC_User_XX to legacy folder

rullzer committed:
Move \OC\User to PSR-4

rullzer committed:
Move OC_User_Database to \OC\User\Database

ChristophWurst committed:
show login error

rullzer committed:
Add \OC\User\Backend

Since some apps (ldap et al) still depend on OC_User_Backend this seemed
like the cleanest approach.

MorrisJobke commented on lib/private/Files/View.php:
```diff
@@ -435,10 +435,12 @@ public function readfilePart($path, $from, $to) {
+			    $end = $to + 1;
+			    while (!feof($handle) && ftell($handle) < $end) {
+				$len = $end-ftell($handle);
+				if ($len > $chunkSize) $len = $chunkSize;
```
Please always use `{` and `}` for `if` statements

MorrisJobke commented on apps/files/ajax/download.php:
```diff
@@ -50,4 +50,17 @@
+	$ranges = array_map('intval', explode('-', substr($_SERVER['HTTP_RANGE'], 6)));
+	if (isset($ranges[0])) {
+	    $server_params['range_from'] = $ranges[0];
+	    if ($ranges[1]) $server_params['range_to'] = $ranges[1];
```
same here

MorrisJobke commented on apps/files_sharing/lib/controllers/sharecontroller.php:
```diff
@@ -484,16 +484,29 @@ public function downloadShare($token, $files = null, $path = '', $downloadStartS
+			$ranges = array_map('intval', explode('-', substr($_SERVER['HTTP_RANGE'], 6)));
+			if (isset($ranges[0])) {
+				$server_params['range_from'] = $ranges[0];
+				if ($ranges[1]) $server_params['range_to'] = $ranges[1];
```
and here

MorrisJobke commented on lib/private/Files/View.php:
```diff
@@ -425,6 +425,34 @@ public function readfile($path) {
+			    $end = $to + 1;
+			    while (!feof($handle) && ftell($handle) < $end) {
+				$len = $end-ftell($handle);
+				if ($len > $chunkSize) $len = $chunkSize;
```
here

MorrisJobke commented on lib/private/files.php:
```diff
@@ -160,15 +169,27 @@ public static function get($dir, $files, $onlyHeader = false) {
+			if ($fileSize > 0) {
+			    $params['range_to'] = $fileSize - 1;
+			    if ($params['range_to'] - $params['range_from'] >= self::DOWNLOAD_PART_MAX_BYTES) 
+				$params['range_to'] = $params['range_from'] + self::DOWNLOAD_PART_MAX_BYTES - 1;
```
here

MorrisJobke commented on lib/private/files.php:
```diff
@@ -178,10 +199,13 @@ private static function getSingleFile($view, $dir, $name, $onlyHeader) {
 		}
-		$view->readfile($filename);
+		if (isset($params['range_from'])) 
+		    $view->readfilePart($filename, $params['range_from'], $params['range_to']);
```
and here

MorrisJobke commented:
Beside my code style comments this looks good and the code works. Now videos could be viewed in the share link page.

MorrisJobke commented:
I like it 👍

lenz1111 committed:
code style fix

lenz1111 committed:
code style fix

lenz1111 committed:
code style fix

lenz1111 committed:
code style fix

ChristophWurst committed:
fix PHPDoc and other minor issues

ChristophWurst committed:
use the query builder instead of raw sql statements

ChristophWurst committed:
delete the token in case an exception is thrown when decrypting the password

f4bsch commented:
This should be used with care, see https://tools.ietf.org/html/rfc7233#section-2.1

The standard permits range header values such as ```bytes=500-600,601-999``` and even ```bytes=0-0,-1```. lenz1111's implementation does not properly handle that, which can cause applications to fail.

For example, Adobe Reader can stop displaying PDF because it uses multi-range requests and gets confused by non-standard range header handling.

You should at least add a file type check and only use the range handling if its a video for example.

rullzer committed:
Move \OCP\Command to PSR-4

lenz1111 commented:
@f4bsch 
You're right. I will add multi-range support in my code - this may be useful not only for movies.

PVince81 commented:
Agreed with range format check, but not file type check. Some people might want to resume downloads of any files using range requests (like download managers do)

MorrisJobke committed:
fix usage of legacy class in LDAP

ChristophWurst committed:
fix login with email

f4bsch commented:
@PVince81 its a dirty workaround to prevent PDF viewers from failing. For PDF documents range downloads are not important, but they might use it for faster access to meta data.  I successfully used this in my WordPress plugin WP-Filebase. Video players usually dont use multi-range downloads, only a single offset byte position to seek.

@lenz1111 Cool, looking forward for that :)

ChristophWurst committed:
create session token for DAV clients (sync clients)

ChristophWurst committed:
use the UID for creating the session token, not the login name

lenz1111 committed:
handling multi-range requests

lenz1111 committed:
handling multi-range requests

lenz1111 committed:
handling multi-range requests

lenz1111 committed:
handling multi-range requests

lenz1111 commented:
@f4bsch Added multi-range support but I haven't found any application that uses multi-range requests. I can simulate it by _curl_   but _curl_ doesn't parse multipart response...

nickvergessen committed:
Correctly register autoloading before install.php and loading commands

PVince81 commented on lib/private/Files/View.php:
```diff
@@ -425,6 +425,36 @@ public function readfile($path) {
+		@ob_end_clean();
+		$handle = $this->fopen($path, 'rb');
+		if ($handle) {
+			if (fseek($handle, $from) === 0) {
```
Some external storages do not support fseek, you'll need a workaround that reads the data up to the file's position.
It might be possible to apply the SeekableWrapper from https://github.com/icewind1991/Streams/blob/master/src/SeekableWrapper.php

@icewind1991 mind writing a clue how to apply the wrapper at this stage ?
Or should we consider adding it at the external storage level instead ?

lenz1111 replied:
@PVince81 I think is it not a good idea to read up to the file position - file could be very large...better is to ignore range request in this case. Can we check that drive support fseek?

f4bsch replied:
@PVince81 ,  @lenz1111 : stream_get_meta_data() should help here. It returns if the file stream is seekable. Depending on its state you should send another HTTP header either
```Accept-Ranges: bytes``` or ```Accept-Ranges: none``` explicitly.

PVince81 replied:
No, `stream_get_meta_data` lies about the stream's seekability when using stream wrappers. Basically a stream wrapper always appears as seekable even if it isn't.
Only `fseek` tells the truth with an error code.

See https://github.com/fruux/sabre-dav/issues/787

lenz1111 replied:
@PVince81 So, I will throw an exception when _fseek_ fail, then remove range request header and return whole file.

PVince81 replied:
Yeah, that sounds good

lenz1111 replied:
fixed

PVince81 commented on lib/private/files.php:
```diff
@@ -49,30 +50,47 @@ class OC_Files {
 
 	const UPLOAD_MIN_LIMIT_BYTES = 1048576; // 1 MiB
 
+	const MULTIPART_BOUNDARY_SEPARATOR = '31b3c516eb6969f9f417f172a315ce8b'; // random string
```
Unlikely to have a collision, but who knows... @LukasReschke what's the best practice here ? Use one of the secure number generators ?

LukasReschke replied:
🙈

LukasReschke replied:
`base64_encode(random_bytes(64))` or so should do. Also,

lenz1111 replied:
code from apps/files_external/3rdparty/Dropbox/API.php:
```
protected function multipartFetch($uri, $file, $filename) {
        /* random string */
        $boundary = 'R50hrfBj5JYyfR3vF3wR96GPCC9Fd2q2pVMERvEaOE3D8LZTgLLbRpNwXek3';

```
As you see this type of boundary is already used in owncloud :)
I can change it to random generated string - but it will be slower version...

LukasReschke replied:
Generating random data is actually pretty cheap overall. And the Dropbox library we use is kinda ... well ... old and needs replacement ;)

PVince81 commented on lib/private/files.php:
```diff
@@ -160,15 +178,63 @@ public static function get($dir, $files, $onlyHeader = false) {
+			$minOffset = 0;
+			$ind = 0;
+
+			foreach ($rArray as $value) {
```
Maybe move this to a separate function to make it easier to write unit tests for it, especially that it has so many if statements. Let us know if you need help. :smile:

LukasReschke commented on lib/private/files.php:
```diff
@@ -49,30 +50,47 @@ class OC_Files {
+	 * @param array $rangeArray ('from'=>int,'to'=>int), ...
 	 */
-	private static function sendHeaders($filename, $name) {
+	private static function sendHeaders($filename, $name, $rangeArray) {
```
`array $rangeArray`

lenz1111 committed:
fix code look

lenz1111 committed:
fix code look

lenz1111 committed:
fix code look

lenz1111 committed:
random multipart boundary

lenz1111 committed:
random multipart boundary

lenz1111 committed:
random multipart boundary

lenz1111 committed:
random multipart boundary

rullzer commented on apps/files_sharing/lib/controllers/sharecontroller.php:
```diff
@@ -484,16 +484,25 @@ public function downloadShare($token, $files = null, $path = '', $downloadStartS
 
 		$this->emitAccessShareHook($share);
 
+		$server_params = array( 'head' => $_SERVER['REQUEST_METHOD'] == 'HEAD' );
```
no... please use the request object to get the header info...

lenz1111 replied:
This is original code in master branch:
`OC_Files::get($originalSharePath, $files_list, $_SERVER['REQUEST_METHOD'] == 'HEAD');`
I only moved this to array....

how can I get this by request object?

rullzer replied:
Aaah missed that...

`$this->request->getMethod()`

should do the trick

rullzer replied:
If it breaks stuff then just leave it... if it was in the original code. Then I'll try to fix it later on.

rullzer commented on apps/files_sharing/lib/controllers/sharecontroller.php:
```diff
@@ -484,16 +484,25 @@ public function downloadShare($token, $files = null, $path = '', $downloadStartS
+		 * Http range requests support
+		 */
+		if (isset($_SERVER['HTTP_RANGE'])) {
+			$server_params['range'] = $_SERVER['HTTP_RANGE'];
```
same here

rullzer replied:
`$this->request->getHeader('Range')`

rullzer commented:
Please rebase. You have edited files that are moved.

JetUni commented:
This contribution is MIT licensed

MorrisJobke commented:
This looks like the rebase somehow failed :(

DeepDiver1975 committed:
Yo-ho-oh - Murder all band 'o pirates

lenz1111 committed:
Merge branch 'share_download_range_requests_support' of https://github.com/lenz1111/core into share_download_range_requests_support

lenz1111 committed:
Merge branch 'share_download_range_requests_support' of https://github.com/lenz1111/core into share_download_range_requests_support

lenz1111 committed:
Merge branch 'share_download_range_requests_support' of https://github.com/lenz1111/core into share_download_range_requests_support

DeepDiver1975 committed:
Merge pull request #24550 from owncloud/kill-en@pirate

Yo-ho-oh - Murder all band 'o pirates

DeepDiver1975 committed:
Merge pull request #24433 from owncloud/user_psr4

lib/private/user to PSR-4

DeepDiver1975 committed:
Merge pull request #24493 from owncloud/command_psr4

Move \OCP\Command to PSR-4

DeepDiver1975 committed:
Merge pull request #24189 from owncloud/pluggable-auth

Pluggable auth

DeepDiver1975 committed:
Merge pull request #24531 from owncloud/psr4-loading-for-install-and-commands

Correctly register PSR-4 autoloading before install.php and loading commands

rullzer committed:
Fix test race condition

E-tag propagation replies on the mtime of the file. Order of events:

1. add file 'foo.txt' with content 'bar'
2. Set mtime to now() - 1
3. Check if etag changed.

Now this goes right often when 1 and 2 happen in the same second.
However imagine

1. add file 'foo.txt' with content 'bar' (at t=0.999)
2. Set mtime to now() - 1 (at t=1.001)

Now the mtime will be set to the same time. Thus not chaning the etag.

DeepDiver1975 committed:
Merge pull request #24568 from owncloud/fix_24566

Fix etag propegation test race condition

lenz1111 commented:
I rebased my branch once again - now should be ok

butonic committed:
allow configuration of memcached options (#23729)

* allow configuration of memcache options

Use production values for memcached as explained in http://apprize.info/php/scaling/15.html

The current implementiation uses ascii based serialization. This PR should reduce traffic to the memcached server.

cc @MorrisJobke @FelixBoehm

* add config sample

* merge config options, throw hint on config error

* fix typo

* fix config sample

nickvergessen committed:
Move Provisioning api to PSR-4 (#24510)

* Move app to PSR-4

* Fix setUp method

* Fix the tests

LukasReschke committed:
Use proper URL generation function (#24576)

Fixes the redirection after login, otherwise `core/files/index` is opened which fails.

nickvergessen committed:
Pass on the error message from the user manager to the UI (#24526)

Commit:
Make permalinks work for trashed files (#24537)

Opening a permalink that points to a trashed file will now display the
file within the "Deleted Files" section in the files web UI.

rullzer commented:
Lenz seems you did not rebase but merged core back into your branch with did not make the conflicts easier to resolve: https://github.com/owncloud/core/pull/24354/commits

See https://git-scm.com/docs/git-rebase

lenz1111 commented:
merged and then used git rebase :)

MorrisJobke commented:
@rullzer @lenz1111 We could squash everything on merge :)

Commit:
[tx-robot] updated from transifex

nickvergessen committed:
Move Encryption app to PSR-4 (#24524)

* Move Encryption to PSR-4

* Move encryption tests to PSR-4

* Fix the tests

nickvergessen committed:
Move dav app to PSR-4 (#24527)

* Move Application to correct namespace and PSR-4 it

* Move dav app to PSR-4

LukasReschke committed:
Do not automatically try to enable index.php-less URLs (#24539)

The current logic for mod_rewrite relies on the fact that people have properly configured ownCloud, basically it reads from the `overwrite.cli.ur
l` entry and then derives the `RewriteBase` from it.

This usually works. However, since the ownCloud packages seem to install themselves at `/owncloud` (because subfolders are cool or so…) _a lot_ of people have just created a new Virtual Host for it or have simply symlinked the path etc.

This means that `overwrite.cli.url` is wrong, which fails hard if it is used as RewriteBase since Apache does not know where it should serve files from. In the end the ownCloud instance will not be accessible anymore and users will be frustrated. Also some shared hosters like 1&1 (because using shared hosters is so awesome… ;-)) have somewhat dubious Apache configurations or use versions of mod_rewrite from the mediveal age. (because updating is money or so…)

Anyhow. This makes this explicitly an opt-in configuration flag. If `htaccess.RewriteBase` is set then it will configure index.php-less URLs, if
admins set that after installation and don't want to wait until the next ownCloud version they can run `occ maintenance:update:htaccess`.

For ownCloud 9.0 we also have to add a repair step to make sure that instances that already have a RewriteBase configured continue to use it by copying it into the config file. That way all existing URLs stay valid. That one is not in this PR since this is unneccessary in master.

Effectively this reduces another risk of breakage when updating from ownCloud 8 to ownCloud 9.

Fixes https://github.com/owncloud/core/issues/24525, https://github.com/owncloud/core/issues/24426 and probably some more.

rullzer committed:
lib/private/ocs to PSR-4 (#24558)

* Move OC_OCS_XX to legacy

* Move \OC\OCS to PSR-4

* OC_OCS_xx to \OC\OCS\xx

* Moved OC_OCS_xx to proper namespace
* OC_OCS_xx is still there for legacy reasons as a wrapper

* No calls to OC_OCS_xx in \OC\OCS\

rullzer committed:
Move \OCP\Comments to PSR-4 (#24565)

nickvergessen committed:
Move files_trashbin to PSR-4 (#24570)

nickvergessen committed:
Move files_versions to PSR-4 (#24571)

nickvergessen committed:
Move the testing app to PSR-4 (#24580)

nickvergessen committed:
Background jobs from info.xml (#24582)

* Move background job registration of Federation to info.xml

* Move background registration of Files app to info.xml

* Move background job registration of files_sharing to info.xml

* Move background job registration of files_trashbin to info.xml

* Move background job registration of files_versions to info.xml

* Move background job registration from user_ldap to info.xml

ChristophWurst committed:
Use CLOB for auth token names to allow long user agent strings

ChristophWurst committed:
run upgrades in incognito mode

nickvergessen committed:
Move Files app to PSR-4 (#24569)

* Move lib/ of Files app to PSR-4

* Move tests to PSR-4

ChristophWurst committed:
Log user out correctly if max session lifetime is reached (#24552)

MorrisJobke committed:
Merge pull request #24563 from owncloud/redis-factory

factor out redis configuration into it's own factory

icewind1991 committed:
only configure the http client once we start using it

rullzer committed:
Move functions.php to legacy

This file should really be properly namespaced etc!

rullzer committed:
Move \OC\Template to PSR-4

ChristophWurst committed:
Show login error message correctly (#24599)

Commit:
Merge branch 'Superduper-Progressbar-branch-thingie' of https://github.com/luckydonald/core into luckydonald-Superduper-Progressbar-branch-thingie

Commit:
[tx-robot] updated from transifex

ChristophWurst committed:
don't spam the log file with failed token validation entries

Commit:
Merge pull request #24605 from owncloud/luckydonald-Superduper-Progressbar-branch-thingie

Luckydonald superduper progressbar branch thingie

Commit:
Merge pull request #24601 from owncloud/template_psr4

Template to PSR-4

rullzer committed:
OC_xx all live in the legacy folder

rullzer committed:
All \OC\ classes are handled by the composer autoloader

rullzer committed:
Fix unit tests

Commit:
Merge pull request #24594 from owncloud/authtoken-name-column-size

Use CLOB for auth token names to allow long user agent strings

Commit:
Merge pull request #24614 from owncloud/token-validation-log

don't spam the log file with failed token validation entries

Commit:
Merge pull request #24600 from owncloud/http-client-lazy-config

only configure the http client once we start using it

Commit:
Merge pull request #22214 from owncloud/ext-better-dropbox-gdrive

Direct links to Google Drive/Dropbox configuration pages from external storage config

Commit:
Merge pull request #22341 from owncloud/pellaeon-pr-breadcrumb-filelist-overlap-drop

Handle case when breadcrumb and filelist item overlap and both receive a drop

Commit:
Merge pull request #24617 from owncloud/autoloader_no_private

our autoloader should not load classes in \OC

Commit:
[tx-robot] updated from transifex

jancborchardt committed:
fix admin share settings sorting, use same order as in sidebar

Commit:
[tx-robot] updated from transifex

Commit:
[tx-robot] updated from transifex

Commit:
[tx-robot] updated from transifex

butonic committed:
Merge pull request #24006 from owncloud/audio_m4a

Adding mimetypes for m4a and m4b

butonic committed:
Merge pull request #24083 from owncloud/fix-race-condition

Read only once

butonic committed:
Merge pull request #24370 from owncloud/iclientservice-dicontainer

add IClientService to DI container

DeepDiver1975 committed:
Fix #24655

Commit:
Merge pull request #24630 from owncloud/one-shall-adjust-the-path

Use proper path

Commit:
Merge pull request #24634 from owncloud/sharesettings-sort

fix admin link share settings sorting, use same order as in sidebar

rullzer committed:
Move \OCP\Contacts to PSR-4 (#24645)

rullzer committed:
Move \OCP\Console to PSR-4 (#24636)

rullzer committed:
Move \OCP\Authentication to PSR-4 (#24632)

rullzer committed:
Move \OCP\Diagnostics to PSR-4 (#24667)

rullzer committed:
Group fixup (#24621)

* Move used OC_Group_xx to \OC\Group

* Add (deprecated) legacy wrapper in legacy, OC_Group_xx

* Replace deprecated use of OC_Group_xx with \OC\Group\xx

Commit:
Merge pull request #24597 from owncloud/run-upgrade-in-incognito-mode

run upgrades in incognito mode

Commit:
Allow chunk GC mtime tolerance for unfinished part chunks

Whenever part chunks are written, every fwrite in the write loop will
reset the mtime to the current mtime. Only at the end will the touch()
operation set the mtime to now + ttl, in the future.

However the GC code is expecting that every chunk with mtime < now are
old and must be deleted. This causes the GC to sometimes delete part
chunks in which the write loop is slow.

To fix this, a tolerance value is added in the GC code to allow for
more time before a part chunk gets deleted.

ChristophWurst committed:
a single token provider suffices

rullzer committed:
\OCP\DB to PSR-4 (#24650)

* Move \OCP\DB to PSR-4

* Fix PHP doc

rullzer committed:
Move \OCP\Files to PSR-4

icewind1991 committed:
use mount providers to setup home storages (#24567)

rullzer committed:
Move \OCP\Encryption to PSR-4 (#24680)

blizzz committed:
Fixed dynamic group ldap access (#23450)

* Fixed dynamic group ldap access

getUserGroups:
Using $userDN instead of $uid to query LDAP
Converting groupDN to group name using API instead of substring
Removing cache processing at the end of the method

* Fixing group handling

added back the cache processing and fixed

* fixed possible indention problem

spaces -> tab conversion

* formatting, white-space changes only

Commit:
[tx-robot] updated from transifex

rullzer committed:
Move \OCP\Http to PSR-4

nickvergessen committed:
Move FederatedFileSharing to PSR-4

Commit:
Merge pull request #24685 from owncloud/files_psr4

Move \OCP\Files to PSR-4

Commit:
Merge pull request #24677 from owncloud/single-token-provider

a single token provider suffices

rullzer committed:
Move \OCP\L10N to PSR-4

Commit:
Merge pull request #24706 from owncloud/http_psr4

Move \OCP\Http to PSR-4

Commit:
Merge pull request #24657 from owncloud/fix-libreoffice-templates

Remove author from libreoffice templates

Commit:
Merge pull request #24691 from owncloud/fixchunkttl

Allow chunk GC mtime tolerance for unfinished part chunks

Commit:
Merge pull request #24701 from owncloud/federatedfilesharing-psr4

Move FederatedFileSharing to PSR-4

Commit:
Merge pull request #24707 from owncloud/l10n_psr4

Move \OCP\L10N to PSR-4

lenz1111 committed:
UnseekableException, remove range request on unseekable storages

Commit:
[tx-robot] updated from transifex

SergioBertolinSG commented:
Not working for me, after applying this patch the server returns 500.

logs:
```
{"reqId":"eYr7dALwWw7kvcTNqdNe","remoteAddr":"HIDDEN_IP","app":"PHP","message":"Cannot redeclare OC\\Files\\View::readfilePart() at \/opt\/owncloud\/lib\/private\/Files\/View.php#467","level":3,"time":"2016-05-19T07:37:25+00:00","method":"GET","url":"\/","user":"--"}
```

lenz1111 committed:
merged changes

lenz1111 commented:
@SergioBertolinSG some files have changed location... - try now fixed changes.
Here is patch for current master branch
[share_download_requests_support.zip](https://github.com/owncloud/core/files/272245/share_download_requests_support.zip)

PVince81 commented:
Ouch, the list of commits is huge now. Looks like a failed rebase.
Let me recreate the PR on a branch in core.

PVince81 commented:
Please follow up here: https://github.com/owncloud/core/pull/24750
The PR is now in the core repo, should also make rebasing easier in case it's needed again.
